### PR TITLE
Feature/redis monitoring

### PR DESCRIPTION
**Pull request by joseluisgraa**

### CODE REVIEW

File: src/controllers/support-access.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import multer from 'multer';
+import redisSvc from '../services/redis';
+import dirs from '../services/dirsDb';
+import dataSvc from '../services/data';
+import ejs from 'ejs';
+
+const upload = multer({ dest: 'uploads/support-access' });
+
+@Controller('support-access')
+class OfstedController {
```
Comment (DrogoNevets) [inline src/controllers/support-access.ts:12]: SupportAccessController

File: src/controllers/support-access.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+        console.log('Users found in the database:');
+        console.log(users);
+
+        users.forEach(async (user: any) => {
+            let uid = user.sub;
+            await redisSvc.revokeSupport(uid);
+        });
+
+        res.redirect('/support-access');
```
Comment (DrogoNevets) [inline src/controllers/support-access.ts:43]: delete file once finished please

File: src/views/remove-support.ejs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<h1 class="govuk-heading-xl">Remove support access</h1>
+
+<form action="/support-access" method="post" enctype="multipart/form-data">
```
Comment (DrogoNevets) [inline src/views/remove-support.ejs:3]: can we add some feedback if redirected to this page, something like "users removed" flash message or some such

**Review by DrogoNevets — CHANGES_REQUESTED** [review]

File: src/controllers/support-access.ts (hunk 1 above)
Comment (joseluisgraa) [reply src/controllers/support-access.ts:12]: ah yes I saw it and forgot to fix it

File: src/controllers/support-access.ts (hunk 2 above)
Comment (joseluisgraa) [reply src/controllers/support-access.ts:43]: ok will do

File: src/views/remove-support.ejs (hunk 3 above)
Comment (joseluisgraa) [reply src/views/remove-support.ejs:3]: I added logging in console, thought about feedback here but didn't know how it works with EJS so left it. Let me have a look.

File: src/controllers/support-access.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import ejs from 'ejs';
+import fs from "fs";
+import RemoveSupportResult, { IRemoveSupportResult } from '../models/removeSupportResult';
+
+const upload = multer({ dest: 'uploads/support-access' });
+
+@Controller('support-access')
+class SupportAccessController {
+
+    private renderView(req: Request, res: Response, opts?: IRemoveSupportResult) {
```
Comment (DrogoNevets) [inline src/controllers/support-access.ts:16]: consider setting default value (shouldnt be needed though) you can do an `if` in the EJS too

File: src/controllers/support-access.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+        ejs.renderFile(
+            './src/views/includes/layout.ejs',
+            {
+                template: '../remove-support.ejs',
+                opts: opts || new RemoveSupportResult()
```
Comment (DrogoNevets) [inline src/controllers/support-access.ts:22]: if default value set the OR isnt needed

File: src/views/remove-support.ejs [hunk 6]
```diff
@@ -0,0 +1,70 @@
+<h1 class="govuk-heading-xl">Remove support access</h1>
+
+<% if (opts.error) { %>
```
Comment (DrogoNevets) [inline src/views/remove-support.ejs:3]: 1) try if(opts?.error)
2) if that fails if(opts && opts.error)

File: src/views/remove-support.ejs [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        <ul class="govuk-list govuk-error-summary__list">
+            <li>
+                <a href="#accountsCSV"><%= opts.error %></a>
+            </li>
+        </ul>
+    </div>
+</div>
+<% } %>
+
+<% if (opts.result) { %>
```
Comment (DrogoNevets) [inline src/views/remove-support.ejs:19]: 1) try if(opts?.result)
2) if that fails if(opts && opts.result)

**Review by DrogoNevets — APPROVED** [review]

**Review by DrogoNevets — APPROVED** [review]

**Pull request merged**